### PR TITLE
FINERACT-977-fix possible NPE in AbstractLoanScheduleGenerator

### PR DESCRIPTION
**Pull request by maektwain**
FINERACT-977

### CODE REVIEW

**Comment by maektwain** [comment]
This could be merged if no errors in  case @vorburger

**Review by vorburger — APPROVED** [review]
LGTM

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java [hunk 1]
```diff
@@ -636,10 +637,12 @@ private void updateAmountsBasedOnCurrentEarlyPayments(final MathContext mc, fina
...
                             .plus(currentPeriodParams.getEarlyPaidAmount()), scheduleParams.getPeriodNumber() + 1, mc));
-            if (loanApplicationTerms.getAmortizationMethod().isEqualInstallment()
-                    && fixedEmiAmount.compareTo(loanApplicationTerms.getFixedEmiAmount()) != 0) {
-                currentPeriodParams.setEmiAmountChanged(true);
-            }
+          if (loanApplicationTerms.getAmortizationMethod().isEqualInstallment()
+            && fixedEmiAmount != null
+            && fixedEmiAmount.compareTo(loanApplicationTerms.getFixedEmiAmount()) != 0) {
+            currentPeriodParams.setEmiAmountChanged(true);
+            currentPeriodParams.plusPrincipalForThisPeriod(currentPeriodParams.getEarlyPaidAmount());
```
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java:644]: Whoa! @maektwain hang on - you probably mistakenly added this extra duplicate line here, right?  :sob:  Or was this intentional, and I'm not getting it? I'm proposing to remove it again in #967... do you agree? (I'm not sure if it's causing FINERACT-855 or if that's an unrelated problem.)

Comment (maektwain) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java:644]: Should I remove this?

Comment (maektwain) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java:644]: ?

Comment (vorburger) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java:644]: This PR is already merged, so I have raised new #967, could you please confirm there if that one looks good to you, if you agree that this was wrong?

**Pull request merged**